diff -dur -x '*~' -x '*.orig' -x '*.rej' freeradius-server-3.0.17.orig/Make.inc.in freeradius-server-3.0.17/Make.inc.in
--- freeradius-server-3.0.17.orig/Make.inc.in	2018-04-17 19:12:27.000000000 +0200
+++ freeradius-server-3.0.17/Make.inc.in	2018-10-09 14:42:09.000000000 +0200
@@ -95,7 +95,7 @@
 
 LOGDIR		= ${logdir}
 RADDBDIR	= ${raddbdir}
-RUNDIR		= ${localstatedir}/run/radiusd
+RUNDIR		= ${localstatedir}/run/freeradius
 SBINDIR		= ${sbindir}
 RADIR		= ${radacctdir}
 LIBRADIUS	= $(top_builddir)/src/lib/$(LIBPREFIX)freeradius-radius.la $(TALLOC_LIBS)
diff -dur -x '*~' -x '*.orig' -x '*.rej' freeradius-server-3.0.17.orig/raddb/radiusd.conf.in freeradius-server-3.0.17/raddb/radiusd.conf.in
--- freeradius-server-3.0.17.orig/raddb/radiusd.conf.in	2018-10-09 14:41:49.000000000 +0200
+++ freeradius-server-3.0.17/raddb/radiusd.conf.in	2018-10-09 14:42:44.000000000 +0200
@@ -68,7 +68,7 @@
 modconfdir = ${confdir}/mods-config
 certdir = ${confdir}/certs
 cadir   = ${confdir}/certs
-run_dir = ${localstatedir}/run/${name}
+run_dir = ${localstatedir}/run/freeradius
 
 # Should likely be ${localstatedir}/lib/radiusd
 db_dir = ${raddbdir}
