diff -dur -x '*~' -x '*.orig' freeradius-server-2.1.4.orig/Make.inc.in freeradius-server-2.1.4/Make.inc.in
--- freeradius-server-2.1.4.orig/Make.inc.in	2009-03-11 03:26:50.000000000 +0100
+++ freeradius-server-2.1.4/Make.inc.in	2009-03-18 09:52:34.000000000 +0100
@@ -48,7 +48,7 @@
 
 LOGDIR		= ${logdir}
 RADDBDIR	= ${raddbdir}
-RUNDIR		= ${localstatedir}/run/radiusd
+RUNDIR		= ${localstatedir}/run/freeradius
 SBINDIR		= ${sbindir}
 RADIR		= ${radacctdir}
 LIBRADIUS	= $(top_builddir)/src/lib/$(LIBPREFIX)freeradius-radius.la
diff -dur -x '*~' -x '*.orig' freeradius-server-2.1.4.orig/raddb/radiusd.conf.in freeradius-server-2.1.4/raddb/radiusd.conf.in
--- freeradius-server-2.1.4.orig/raddb/radiusd.conf.in	2009-03-18 09:52:24.000000000 +0100
+++ freeradius-server-2.1.4/raddb/radiusd.conf.in	2009-03-18 09:53:13.000000000 +0100
@@ -66,7 +66,7 @@
 
 #  Location of config and logfiles.
 confdir = ${raddbdir}
-run_dir = ${localstatedir}/run/${name}
+run_dir = ${localstatedir}/run/freeradius
 
 # Should likely be ${localstatedir}/lib/radiusd
 db_dir = ${raddbdir}
