diff -dur -x '*~' -x '*.orig' -x '*.rej' freeradius-server-3.0.17.orig/raddb/radiusd.conf.in freeradius-server-3.0.17/raddb/radiusd.conf.in
--- freeradius-server-3.0.17.orig/raddb/radiusd.conf.in	2018-04-17 19:12:27.000000000 +0200
+++ freeradius-server-3.0.17/raddb/radiusd.conf.in	2018-10-09 14:33:41.000000000 +0200
@@ -398,8 +398,8 @@
 	#  member.  This can allow for some finer-grained access
 	#  controls.
 	#
-#	user = radius
-#	group = radius
+	user = radius
+	group = radius
 
 	#  Core dumps are a bad thing.  This should only be set to
 	#  'yes' if you're debugging a problem with the server.
